packages/flutter/lib/src/services/image_stream.dart

```diff
@@ -365,6 +365,17 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
 
   void _handleCodecReady(ui.Codec codec){
     _codec = codec;
+    if (_codec == null) {
+      // This matches the behavior of OneFrameImageStreamCompleter in the case
+      // we failed to fetch the data.
+      // This can currently happen when loading image data in Image.network
+      // fails. We should probably assert that this doesn't happen once we
```

Comment:
Looks like this is a bug, and NetworkImage._loadAsync should be throwing rather than returning null when it fails.

Reply:
done